portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,49 +31,81 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+
+@WebServlet("/new-post")
 public class DataServlet extends HttpServlet {
  private ArrayList<String> check; 
  public void init() {
+
     check = new ArrayList<String>();
      check.add("Mood");
   check.add("Summer");
     check.add("Style");
   check.add("WFH");
     check.add("Experience");
   check.add("Awesome!");
+
 }
 
+ 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-  //Initializing a gson instance
-  Gson gson = new Gson();
-  //Converting the Arraystring to a Json. 
-    String json = gson.toJson(check);
-   
-    response.setContentType("text/html;");
-    response.getWriter().println(json);
+  response.setContentType("text/html;");
+    response.setContentType("application/json;");
     
+    Query query = new Query("Messages").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    PreparedQuery results = datastore.prepare(query); 
+
+ 
+
+    
+ 
+    for (Entity entity : results.asIterable()) {
+         long id = entity.getKey().getId();
+        response.getWriter().println(entity.getProperty("first_name"));
+        response.getWriter().println(entity.getProperty("last_name"));
+        response.getWriter().println(entity.getProperty("email"));
+        response.getWriter().println(entity.getProperty("comment"));
+        response.getWriter().println(entity.getProperty("timestamp"));
+        response.getWriter().println("\n");
+        
+       
+        
+     
+    }
+
+  Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    //response.getWriter().println(gson.toJson(tasks));
+
   }
 
+
  @Override
+
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+  
       //Fetching the parameters from html form in index.html
-        String first_name = getParameter(request, "first_name", "");
-    String last_name = getParameter(request, "last_name", "");
-        String email = getParameter(request, "email", "");
-    String comment = getParameter(request, "comment", "");
+    String first_name = request.getParameter("first_name");
+    String last_name = request.getParameter("last_name");
+    String email = request.getParameter("email");
+    String comment = request.getParameter("comment");
```

Comment:
since you don't re-use these variable names, you can just inline below as "taskEntity.setProperty("first_name", request.getParameter("first_name")".

Reply:
Since the next step will probably be using this variables in a different method, I better keep it as it is for now.